src/button-group/stateful-container.js

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import * as React from 'react';
+
+import {MODE} from './constants.js';
+
+import type {StatefulContainerPropsT, StateT} from './types.js';
+
+export default class StatefulButtonGroup extends React.Component<
+  StatefulContainerPropsT,
+  StateT,
+> {
+  constructor(props: StatefulContainerPropsT) {
+    super(props);
+
+    this.state = {
+      selected: props.selected ? [].concat(props.selected) : [],
```

Comment:
We use an `initialState: {selected}` API in other stateful components. That sometimes creates some confusion around passing `selected` for a stateless vs `initialState: {selected}` for a stateful components. But the idea of an `initialState` prop was to avoid another often confusion around 'updated the props passed but the `selected` not updated'. 
Should we be consistent with `initialState` prop for stateful components?